plugins/aiops/pkg/gateway/metricai.go

```diff
@@ -0,0 +1,363 @@
+	},
+}
+
+type RequestError struct {
+	StatusCode int
+	Err        error
+}
+
+func (r *RequestError) Error() string {
+	return r.Err.Error()
+}
+
+func (p *AIOpsPlugin) CreateGrafanaDashboard(ctx context.Context, jobRunId *metricai.MetricAIId) (*metricai.MetricAIAPIResponse, error) {
+	// dashboardJson is generated in the python service. This function simply create a GrafanaDashboard resource with it and apply it
+	res, err := p.GetJobRunResult(ctx, jobRunId)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "Failed to Get JobRunRes for metricAI: %v", err)
+	}
+	dashboardJson := string(res.JobRunResultDetails)
+	dashboardName := strings.ToLower(strings.ReplaceAll(res.JobRunId, jobRunDelimiter, ""))
+	grafanaDashboards := []*grafanav1alpha1.GrafanaDashboard{
+		{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      dashboardNamePrefix + dashboardName,
+				Namespace: os.Getenv("POD_NAMESPACE"),
+				Labels:    dashboardSelector.MatchLabels,
+			},
+			Spec: grafanav1alpha1.GrafanaDashboardSpec{
+				Json: dashboardJson,
+			},
+		},
+	}
+	for _, dashboard := range grafanaDashboards {
+		err := p.k8sClient.Create(ctx, dashboard)
+		if err != nil {
+			return nil, status.Errorf(codes.Internal, "Error Creating Dashboard for metricAI: %v", err)
+		}
+	}
+	return &metricai.MetricAIAPIResponse{SubmittedTime: time.Now().String(), Description: dashboardJson, Status: "Success"}, nil
+}
+
+func (p *AIOpsPlugin) DeleteGrafanaDashboard(ctx context.Context, jobRunId *metricai.MetricAIId) (*metricai.MetricAIAPIResponse, error) {
+	// delete the grafanadashboard resource for the given jobrun id
+	res, err := p.GetJobRunResult(ctx, jobRunId)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "Failed to Get JobRunRes for metricAI: %v", err)
+	}
+	dashboardJson := res.JobRunResultDetails
+	dashboardName := strings.ToLower(strings.ReplaceAll(res.JobRunId, jobRunDelimiter, ""))
+	grafanaDashboards := []*grafanav1alpha1.GrafanaDashboard{
+		{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      dashboardNamePrefix + dashboardName,
+				Namespace: os.Getenv("POD_NAMESPACE"),
+				Labels:    dashboardSelector.MatchLabels,
+			},
+			Spec: grafanav1alpha1.GrafanaDashboardSpec{
+				Json: dashboardJson,
+			},
+		},
+	}
+	for _, dashboard := range grafanaDashboards {
+		err := p.k8sClient.Delete(ctx, dashboard)
+		if err != nil {
+			return nil, status.Errorf(codes.Internal, "Error Deleting Dashboard for metricAI: %v", err)
+		}
+	}
+	return &metricai.MetricAIAPIResponse{SubmittedTime: time.Now().String(), Description: dashboardJson, Status: "Success"}, nil
+}
+
+func (p *AIOpsPlugin) ListClusters(ctx context.Context, _ *emptypb.Empty) (*metricai.MetricAIIdList, error) {
+	// For the UI to list clusters. Returns cluster_id
+
+	timeout := 10 * time.Second
+	ctxca, cancel := context.WithTimeout(ctx, timeout)
+	defer cancel()
+	httpClient := &http.Client{Timeout: timeout}
```

Comment:
Probably won't be an issue but for scaling I think it's probably better to create the httpClient in the struct and reuse the same one.

Reply:
good point. fixed